csrc/quantization/gptq_marlin/gptq_marlin.cu

```diff
@@ -1170,11 +1202,40 @@ __global__ void Marlin(
         }
       }
     }
+
+    else if constexpr (has_zp && is_zp_float) {
+      int pipe = full_pipe % stages;
+
+      if constexpr (group_blocks != -1) {
+        if constexpr (group_blocks >= thread_k_blocks) {
+          int4* sh_zp_stage =
+              sh_zp + zp_sh_stage * ((group_blocks / thread_k_blocks) *
+                                     (pipe / (group_blocks / thread_k_blocks)));
+          reinterpret_cast<int4*>(&frag_zpf[k % 2])[0] = sh_zp_stage[zp_sh_rd];
+        } else {
+          int warp_id = threadIdx.x / 32;
+          int n_warps = thread_n_blocks / 4;
+
+          int warp_row = warp_id / n_warps;
+
+          int cur_k = warp_row * 16;
+          cur_k += k_iter_size * (k % b_sh_wr_iters);
+
+          int k_blocks = cur_k / 16;
+          int cur_group_id = k_blocks / group_blocks;
```

Comment:
@ElizaWszola have you seen any compile warnings from this? 

I don't see any in the logs but notice the compiler pragmas I added around lines 1190-1193. The bogus warnings might have disappeared possibly due to a compiler upgrade, but I think it's best to keep them around for consistency and to make sure folks on older compilers don't get spammed.

Reply:
Thanks